features/0593-json-ld-cred-attach/README.md

```diff
@@ -0,0 +1,206 @@
+# Aries RFC 0593: JSON-LD Credential Attachment format for requesting and presenting credentials
+
+- Authors: Timo Glastra (Animo Solutions)
+- Status: [PROPOSED](/README.md#proposed)
+- Since: 2021-02-24
+- Status Note:
+- Supersedes:
+- Start Date: 2021-02-17
+- Tags: [feature](/tags.md#feature), [protocol](/tags.md#protocol), [credentials](/tags.md#credentials), [test-anomaly](/tags.md#test-anomaly)
+
+## Summary
+
+This RFC registers an attachment format for use in the [issue-credential V2](../0453-issue-credential-v2/README.md) protocol based on JSON-LD credentials with [Linked Data Proofs](https://w3c-ccg.github.io/ld-proofs/) from the [VC Data Model](https://www.w3.org/TR/vc-data-model/#linked-data-proofs).
+
+It defines a minimal set of parameters needed to create a common understanding of the verifiable credential to issue. It is based on version [1.0 of the Data Model](https://www.w3.org/TR/vc-data-model/) which is a W3C recommendation since 19 November 2019.
+
+## Motivation
+
+The Issue Credential protocol needs an attachment format to be able to exchange JSON-LD credentials with Linked Data Proofs. It is desirable to make use of specifications developed in an open standards body, such as the [Credential Manifest](https://identity.foundation/credential-manifest/) for which the attachment format is described in [RFC 0511: Credential-Manifest Attachment format](../0511-dif-cred-manifest-attach/README.md). However, the _Credential Manifest_ is not finished and ready yet, and therefore there is a need to bridge the gap between standards.
+
+## Tutorial
+
+Complete examples of messages are provided in the [reference section](#reference).
+
+## Reference
+
+### `ld-proof-vc-proposal` attachment format
+
+Format identifier: `aries/ld-proof-vc-proposal@v1.0`
+
+The credential proposal allows the holder to initiate a credential exchange without fully specifying the contents of the credential. If all properties of the credential are already known, the protocol can also be initiated with a credential request. While the credential detail requires all fields to be present, the proposal allows for fields to be omitted.
+
+The below `propose-credential` message shows an example of a credential proposal with the `@context`, `type`, `credentialSubject` and `credentialSubject` present. The issuer can then send a credential detail in a credential offer message adding extra properties required for issuance such as `issuer`, `credentialStatus.type`, `proofType`, etc..
```

Comment:
```suggestion
The below `propose-credential` message shows an example of a credential proposal with the `@context`, `type`, `credentialSubject` and `credentialSchema` present. The issuer can then send a credential detail in a credential offer message adding extra properties required for issuance such as `issuer`, `credentialStatus.type`, `proofType`, etc..
```

Reply:
I am curious about the inclusion of the `credentialSchema` property.

Since the VC data model is heavily biased towards JSONLD, we have been using the top-level JSONLD `@context` to determine the vocabulary (aka. "schema") of the credential. Is the intent here to use `credentialSchema` for just that?

Reply:
I think the `schema.uri` from PE is a good reason for `credentialSchema` to be included. This allows to filter credentials by their `credentialsSchema` in PE.

Looking at your implementation that's what you're doing already:

https://github.com/hyperledger/aries-framework-go/blob/cb1ce0da0c3b5d76db78444719268b2d00a4d2f9/pkg/doc/presexch/definition.go#L358

Reply:
We have inconsistent behaviour in AFGO that we need to fix:

* Match(): used by the Verifier. Matches `schema.uri` against `@context`
* CreateVP(): used by the Holder. Matches `schema.uri` against `credentialSchema.id`

Reply:
We agreed to use the full JSONLD type URI as shown in an example [here](https://github.com/decentralized-identity/presentation-exchange/issues/134#issuecomment-721624167)